lib/ecto/enum.ex

```diff
@@ -4,19 +4,23 @@ defmodule Ecto.Enum do
 
   `Ecto.Enum` must be used whenever you want to keep atom values in a field.
   Since atoms cannot be persisted to the database, `Ecto.Enum` converts them
-  to string when writing to the database and converts them back to atoms when
-  loading data. It can be used in your schemas as follows:
+  to a string or an integer when writing to the database and converts them back
+  to atoms when loading data. It can be used in your schemas as follows:
```

Comment:
Maybe we could generalize and support anything?

For instance, a list of [:a, :b, :c] is equal to [a: "a", b: "b", c: "c"], and maybe one could use something like [a: "b", b: "c", c:   "d"]. Seems like a more coherent API for me.

Reply:
Yes, let's go down this route. It may simplify the code too. Basically, if a list of atom is given, convert it to: `Enum.map(list, fn {k, to_string(k)} end)` and write the rest of the code assuming `values` is always a keyword. Note we don't need to validate the values, let's just assume they are valid.

Reply:
Generalizing things and allowing for custom string mappings is a great idea.

I have added support for things like ```[a: "b", b: "c", c: "d"]``` in a00cef6b0d2d40e4f0d4aa6c9e20b202e7ed3105

@josevalim I have changed the list of atoms to a keyword list as you suggested and it cleaned up the code nicely :-)